test/src/test/scala/shapeless/datatype/test/Records.scala

```diff
@@ -29,7 +29,7 @@ object Records {
                    longFieldR: List[Long], doubleFieldR: List[Double], stringFieldR: List[String])
   case class Nested(longField: Long, longFieldO: Option[Long], longFieldR: List[Long],
                     mixedField: Mixed, mixedFieldO: Option[Mixed], mixedFieldR: List[Mixed])
-  case class Seqs(array: Array[Int], list: List[Int], vector: Vector[Int])
+  case class Iterables(/*array: Array[Int], */list: List[Int], vector: Vector[Int], set: Set[Int])
```

Comment:
I really can't seem to work out why but it doesn't compile when that array field is present, it just can't seem to derive the instances. Which seems very strange since there are other classes in this file which have arrays and their instances are derived without any problems.

Reply:
@nevillelyh any ideas?

Reply:
I think it's because `Array` is a Java array and doesn't satisfy the `<: Iterable[_]` bound. I don't think that bound is necessary since the implicit `toIterable: S[V] => Iterable[V]` should cover it.